### PR TITLE
Pre-commit

### PR DESCRIPTION
**Pull request by lequ0n**

### CODE REVIEW

File: .pre-commit-config.yaml [hunk 1]
```diff
@@ -0,0 +1,16 @@
...
+      - id: trailing-whitespace
+      - id: check-case-conflict
+      - id: check-docstring-first
+      - id: double-quote-string-fixer
+      - id: detect-private-key
+
+  - repo: https://github.com/google/yapf
+    rev: v0.40.2
+    hooks:
+      - id: yapf
```
Comment (grzanka) [inline .pre-commit-config.yaml:15]: in other projects we use black, can you switch to it ?
https://black.readthedocs.io/en/stable/integrations/source_version_control.html

**Review by grzanka — CHANGES_REQUESTED** [review]
see comments

File: .pre-commit-config.yaml (hunk 1 above)
Comment (lequ0n) [reply .pre-commit-config.yaml:15]: Should we really switch to black? Both yaptide and pymchelper use yapf. Also, in my opinion, the line length limitation in black often makes the code less readable.

Comment (lequ0n) [reply .pre-commit-config.yaml:15]: Example:

YAPF:
```python
if "worldZone" in zones_json and "averageIonisationPotential" in zones_json["worldZone"].get(
            "materialPropertiesOverrides", []):
        ionisation_potentials.append(
            IonisationPotential(
                material_name=materials[zones_json["worldZone"]["materialUuid"]].fluka_name,
                ionisation_potential=zones_json["worldZone"]["materialPropertiesOverrides"]
                ["averageIonisationPotential"],
            ))
```									

BLACK:

```python
if "worldZone" in zones_json and "averageIonisationPotential" in zones_json[
        "worldZone"
    ].get("materialPropertiesOverrides", []):
        ionisation_potentials.append(
            IonisationPotential(
                material_name=materials[
                    zones_json["worldZone"]["materialUuid"]
                ].fluka_name,
                ionisation_potential=zones_json["worldZone"][
                    "materialPropertiesOverrides"
                ]["averageIonisationPotential"],
            )
        )
```

Comment (grzanka) [reply .pre-commit-config.yaml:15]: You are correct, yapf is used there. But don't forget to configure line length in yapf. 
See https://github.com/yaptide/yaptide/blob/0d0f76a178268cae75bfe966d37b1561503e6dea/setup.cfg#L7

Both yapf and black use ~80 characters by default. And both can be configured to accept longer lines

**Pull request merged**